internal/plugin/connectors/tcp/template/plugin.go

```diff
@@ -0,0 +1,49 @@
+package template
+
+import (
+	"net"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+// NewConnector returns a tcp.Connector which returns an authenticated connection
+// to a target service for each incoming client connection.
+// It is a required method on the tcp.Plugin interface.
+// The single argument passed in is of type connector.Resources. It contains
+// connector-specific config and a logger.
+func NewConnector(conRes connector.Resources) tcp.Connector {
+	connectorFunc := func(
+		clientConn net.Conn,
+		credentialValuesByID connector.CredentialValuesByID,
+	) (backendConn net.Conn, err error) {
+		// singleUseConnector is responsible for generating the authenticated connection
+		// to the target service for each incoming client connection
+		singleUseConnector := &SingleUseConnector{
+			logger: conRes.Logger(),
+		}
+
+		return singleUseConnector.Connect(clientConn, credentialValuesByID)
+	}
+
+	return tcp.ConnectorFunc(connectorFunc)
+}
+
+// PluginInfo is required as part of the Secretless plugin spec. It provides
+// important metadata about the plugin.
+func PluginInfo() map[string]string {
+	// TODO: fill in the map according to
+	// https://github.com/cyberark/secretless-broker/blob/master/pkg/secretless/plugin/connector/README.md#plugininfo
+	return map[string]string{
+		"pluginAPIVersion": "",
+		"type":             "",
```

Comment:
This is standardized for tcp connectors: https://github.com/cyberark/secretless-broker/blob/master/internal/plugin/connectors/tcp/pg/plugin.go#L35